src/Provider/ActiveMQ.php

```diff
@@ -24,53 +26,49 @@ class ActiveMQ extends AbstractBaseProvider
      */
     private $client;
 
+    /**
+     * @var bool
+     */
+    private $statisticsPluginEnabled;
+
+    /**
+     * @var bool
+     */
+    private $persistentMessage;
+
     /**
      * ActiveMQ constructor.
      *
      * @param string $brokerUri
      * @param string $login
      * @param string $password
+     * @param bool   $persistentMessage
+     * @param bool   $statisticsPluginEnabled
      */
-    public function __construct($brokerUri, $login, $password)
+    public function __construct($brokerUri, $login, $password, $persistentMessage = false, $statisticsPluginEnabled = false)
     {
         $connection = new Connection($brokerUri);
-        $connection->setReadTimeout(5, 0);
 
         $this->client = new Client($connection);
         $this->client->setLogin($login, $password);
         $this->client->setSync(true);
 
         $this->stomp = new StatefulStomp($this->client);
+
+        $this->statisticsPluginEnabled = $statisticsPluginEnabled;
+        $this->persistentMessage = $persistentMessage;
     }
 
     /**
      * {@inheritdoc}
      */
     public function count($queueName)
     {
-        $tempQueueName = sprintf("/temp-queue/%s", uniqid());
-
-        $this->client->sendFrame(new Message(
-            "",
-            [
-                "destination" => $tempQueueName,
-            ]
-        ), true);
-
-        $this->client->sendFrame(new Message(
-            "",
-            [
-                "reply-to"    => $tempQueueName,
-                "destination" => "ActiveMQ.Statistics.Destination.".$queueName,
-            ]
-        ), true);
-
-        $subscriptionId = $this->stomp->subscribe($tempQueueName);
+        if (false === $this->statisticsPluginEnabled) {
+            throw new NotImplementedFeatureException();
```

Comment:
Can you add a message that say to use the statistics plugin to use this method ?

Reply:
Done